Functions/ImageProcessing/xASL_im_Lesion2Mask.m

```diff
@@ -198,17 +198,18 @@
     OutIm{iIm} = xASL_vis_CreateVisualFig(x, {x.P.Pop_Path_rT1 VisualizeImage==iIm}, [], [0.75 0.35], [], {x.S.gray x.S.colors_ROI{iIm}});
 end
 
-% Initialize final image
-OutImFinal = zeros(size(OutIm{1}));
-OutImFinal(OutIm{1}==OutIm{2}) = OutImFinal(OutIm{1}==OutIm{2});
+% Initialize final image - make sure it has the proper dimensions when an individual image is returned empty
+OutImFinal = zeros(max(cellfun(@(x) size(x,1),OutIm)), max(cellfun(@(x) size(x,2),OutIm)), max(cellfun(@(x) size(x,3),OutIm)));
```

Comment:
I would replace cellfun(@(x) by cellfun(@(y) as we use x as a Table throughout ExploreASL. It might work this way but never hurts to be careful. Also, I use y throughout, so we are more consistent.

Reply:
It definitively isn't a problem from the technical standpoint. But no prob to change to improve readability. Done